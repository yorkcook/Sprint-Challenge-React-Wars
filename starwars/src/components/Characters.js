import React, { useState, useEffect } from "react";

import axios from "axios";
import Element from "./Element";

const Characters = () => {
  const [charactersAll, setCharactersAll] = useState([]);

  useEffect(() => {
    axios
      .get(`https://swapi.co/api/people/`)
      .then(res => {
        console.log("res", res);
        setCharactersAll(res.data.results);
      })
      .catch(err => console.log("That's not a planet, it's a space station!"));
  }, []);

  return (
    <div>
      <h1>Characters</h1>
      {charactersAll.map(characters => (
        <Element characters={characters} />
      ))}
    </div>
  );
};

export default Characters;
