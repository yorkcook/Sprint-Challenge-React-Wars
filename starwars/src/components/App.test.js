import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import renderer from "react-test-renderer";
import Characters from "./Characters";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<App />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Characters />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<Characters />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Characters />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
