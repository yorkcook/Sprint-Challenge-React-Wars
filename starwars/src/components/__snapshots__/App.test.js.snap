// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="Header"
  >
    React Wars
  </h1>
  <div>
    <h1>
      Characters
    </h1>
  </div>
</div>
`;

exports[`<Characters /> should match snapshot 1`] = `
<div>
  <h1>
    Characters
  </h1>
</div>
`;
